support reference bundle in build-on-commit action

kwmonroe opened the pull request:
Add an optional field so users can specify the reference bundle to use when
testing their charm. If set, the action will build the charm locally and
deploy it. It will then deploy the given reference bundle. Bundletester
will re-use the deployed local charm in place of the one specified in
bundle.yaml.

johnsca commented on templates/BuildMyCharm/config.xml:
```diff
@@ -51,7 +51,14 @@ charm build
+then
+    bundletester -e {{controller}}:test-$BUILD_NUMBER -vFt /tmp/builds/{{charmname}}
+else
+    juju deploy /tmp/builds/{{charmname}}
```
This should ensure it deploys to the correct controller.  It should also get the service name from the bundle to ensure that the locally built charm is used.

kwmonroe commented:
@johnsca I think I addressed your concerns.  This now deploys the charm -- using the service name from the bundle -- to the explicit model.

Be nice if it worked.  I'm pushing this up broken so @ktsakalozos can take over.  Running the action results in an import error for theblues which I can't figure out:

```
2016-12-12 23:41:52 INFO build-on-commit Traceback (most recent call last):
2016-12-12 23:41:52 INFO build-on-commit   File "/var/lib/juju/agents/unit-cwr2-0/charm/actions/build-on-commit", line 5, in <module>
2016-12-12 23:41:52 INFO build-on-commit     from cwrhelpers import app_from_bundle, fail_action
2016-12-12 23:41:52 INFO build-on-commit   File "/var/lib/juju/agents/unit-cwr2-0/charm/actions/cwrhelpers.py", line 9, in <module>
2016-12-12 23:41:52 INFO build-on-commit     from theblues import charmstore
2016-12-12 23:41:52 INFO build-on-commit ImportError: No module named 'theblues'
```

This is a subordinate charm that installs theblues into a venv (and yes, it's there).  Do actions need to do something special to work in the venv?  I trust kostas will tell me soon enough ;)

johnsca commented:
@kwmonroe @ktsakalozos We need a helper to activate the venv for actions, so I created juju-solutions/layer-basic#80 to refactor that bit out.  The doc string has example usage.
